compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java

```diff
@@ -1726,22 +1771,81 @@ private void addToBlockedList(BLangInvocation invocationExpr) {
         this.taintedStatus = TaintedStatus.UNTAINTED;
     }
 
+    private void combinedParameterTaintedStatus(List<TaintedStatus> combinedArgTaintedStatus,
+                                                List<Boolean> argTaintedStatus) {
+        if (argTaintedStatus != null) {
```

Comment:
What would be the behaviour if argTaintedStatus is null? Also I would rather fail fast here, inverting this if